source/common/network/raw_buffer_socket.cc

```diff
@@ -18,6 +18,7 @@ IoResult RawBufferSocket::doRead(Buffer::Instance& buffer) {
   do {
     // 16K read is arbitrary. TODO(mattklein123) PERF: Tune the read size.
     int rc = buffer.read(callbacks_->fd(), 16384);
+    int error = errno; // Latch errno before any logging calls can overwrite it.
```

Comment:
Nit: make these all `const`, less likely that we will forget it after latching ;)

Reply:
Darn it!  
Days since alyssar@ forgot const checks -> 0 :-P